lib/ACL/ACLManager.php

```diff
@@ -98,13 +99,22 @@ private function getRules(array $paths, bool $cache = true): array {
 	 * @return string[]
 	 */
 	private function getRelevantPaths(string $path): array {
-		$paths = [$path];
+		$paths = [];
+		$fromTrashbin = str_starts_with($path, '__groupfolders/trash/');
+		if ($fromTrashbin) {
+			$rootName = explode('/', $path, 5)[3];
+			$rootName = substr($rootName, 0, strrpos($rootName, '.d'));
+		}
 		while ($path !== '') {
+			$paths[] = $path;
 			$path = dirname($path);
-			if ($path === '.' || $path === '/') {
+			if ($fromTrashbin && ($path === '__groupfolders/trash')) {
+				$trashItemRow = $this->trashManager->getTrashItemByFileName($rootName);
+				$path = dirname('__groupfolders/' . $trashItemRow['folder_id'] . '/' . $trashItemRow['original_location']);
```

Comment:
Will not work if `original_location` is renamed, right?

Reply:
Indeed.
But then not sure how to fix that, it means we need to duplicate ACLs when things are trashed?
Or we need to rename `original_location` when parent is renamed.

1. Groupfolder1/subfolder/A.txt is deleted
2. Groupfolder1/subfolder is renamed to Groupfolder1/subfolder2
3. A.txt is restored
4. What is expected? What is currently happening?

Reply:
After discussing that with designers, the link to original_location should follow renames so that restoring restores in the renamed folder.
So this will be fixed in a follow-up PR.